grouper/fe/handlers.py

```diff
@@ -717,6 +718,136 @@ def get(self, group_id=None, name=None):
         )
 
 
+class AuditsComplete(GrouperHandler):
+    def post(self, audit_id):
+        # TODO: Make it so only certain users can start audits
+        user = self.get_current_user()
+
+        audit = self.session.query(Audit).filter(Audit.id == audit_id).one()
+        if audit.complete:
+            return self.redirect("/groups/{}".format(audit.group.name))
+
+        edges = {}
+        for argument in self.request.arguments:
+            if argument.startswith('audit_'):
+                edges[int(argument.split('_')[1])] = self.request.arguments[argument][0]
+
+        for member in audit.my_members():
+            if member.id in edges:
+                if edges[member.id] in AUDIT_STATUS_CHOICES:
+                    member.status = edges[member.id]
+
+        self.session.commit()
+
+        # Now if it's completable (no pendings) then mark it complete, else redirect them
+        # to the group page.
+        if not audit.completable:
+            return self.redirect('/groups/{}'.format(audit.group.name))
+
+        # Complete audits have to be "enacted" now. This means anybody marked as remove has to
+        # be removed from the group now.
+        for member in audit.my_members():
+            if member.status == "remove":
+                audit.group.revoke_member(self.current_user, member.member,
+                                          "Revoked as part of audit.")
+                AuditLog.log(self.session, self.current_user.id, 'remove_member',
```

Comment:
Should we record "Membership confirmed as part of audit." for members who do belong?  Mostly we'd expect audits to find "things are as they should be", at least at a properly-managed institution.